src/library/string.cpp

```diff
@@ -29,16 +29,18 @@ static expr * g_fin_mk               = nullptr;
 
 expr from_string_core(std::string const & s);
 
-static void display_char_literal_core(std::ostream & out, unsigned char c, bool in_string) {
+static void display_char_literal_utf8(std::ostream & out, unsigned char c, bool in_string) {
     if (c == '\n') {
         out << "\\n";
     } else if (c == '\t') {
         out << "\\t";
+    } else if (c == '\\') {
+        out << "\\\\";
     } else if (in_string && c == '\"') {
         out << "\\\"";
     } else if (!in_string && c == '\'') {
         out << "\\'";
-    } else if (32 <= c && c <= 126) {
+    } else if (c >= 32 && c != 127) {
```

Comment:
It would of course be preferable if we escaped any non-printable Unicode character, but that is not exactly trivial to decide.